### PR TITLE
Resolved issue for Customer address is duplicated after setBillingAddressOnCart GraphQL mutation

### PR DESCRIPTION
**Pull request by Aniket10Nov**
### Description (*)

Resolved issue for Customer address is duplicated after setBillingAddressOnCart GraphQL mutation. If the customer set billing address is the same as the shipping address then it won't save as a new address.

### Related Pull Requests

#26884
### Fixed Issues (if relevant)

1. magento/magento2#26884 Customer address is duplicated after setBillingAddressOnCart GraphQL mutation.

### Manual testing scenarios (*)

Do all steps via GQL:

1. Authorize ( generateCustomerToken mutation)
2. Create cart (customerCart query)
3. Add product to cart (addSimpleProductsToCart mutation)
4. Set billing address (setBillingAddressOnCart mutation).
```
Important: save_in_address_book: true
input:{ 
   cart_id:"cart_id"   
   billing_address:{ 
      address:{ 
      ...
         save_in_address_book:true
      }
      same_as_shipping:true
   }
}
```

5. Set shipping method and payment method to the cart.
6. Place order (placeOrder mutation)

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php [hunk 1]
```diff
@@ -90,7 +90,7 @@ public function execute(ContextInterface $context, CartInterface $cart, array $b
             );
         }
 
-        $billingAddress = $this->createBillingAddress($context, $customerAddressId, $addressInput);
+        $billingAddress = $this->createBillingAddress($context, $customerAddressId, $addressInput,$sameAsShipping);
```
Comment (rogyar) [inline app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php]: Please, make sure you preserve the PSR code style and add whitespace before the argument

**Review by rogyar — CHANGES_REQUESTED** [review]

**Comment by rogyar** [comment]
Hi @Aniket10Nov. According to the new [Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html), all new changes should be covered by automated tests. Since we don't have a GraphQl query for checking customer addresses, we are not able to cover this case using an API-functional test. 
But we have a possibility to perform a GraphQl request for setting the quote address and then use the repository or resource model to assert customer addresses using an integration test. Please, refer to the following test as an example.

**Comment by engcom-Charlie** [comment]
Hi @Aniket10Nov it is M2 core issue, please see https://github.com/magento/magento2/issues/22596.

**Comment by prabhuram93** [comment]
Hi @Aniket10Nov , @rogyar! Our internal team has started working on this in the scope of MC-31586

**Comment by lenaorobei** [comment]
Covered by tests internally.

**Pull request merged**